### DISCUSSION TITLE
`bbox_params` in nested `Compose`s

### DISCUSSION THREAD

**Discussion by AdeelH** [albumentations-team/albumentations#1411] [Q&A] [answered]
It seems like when you have nested `Compose`s, you only need to specify `bbox_params` in the outer most `Compose`. For example, the following transform works without problems:
```py
bbox_params = A.BboxParams(format='albumentations', label_fields=['category_id'])
tf = A.Compose([
    A.Compose([
        A.Compose([
            A.ShiftScaleRotate(always_apply=True),
        ]),
    ]),
], bbox_params=bbox_params)
```

This behavior does not seem to be documented anywhere. Can you confirm that this is the intended behavior and that it always holds?

**Comment by Dipet** [comment, marked as answer]
Yes, it is not documented, but it is works. We do not recomend use `Compose` as inner transform. For this purpose we have [`Sequential`](https://albumentations.ai/docs/api_reference/core/composition/#albumentations.core.composition.Sequential)
